tsfresh/utilities/distribution.py

```diff
@@ -0,0 +1,252 @@
+        """
+        if data_length is None:
+            data_length = len(data)
+
+        if not chunk_size:
+            chunk_size = self._calculate_best_chunksize(data_length)
+
+        chunk_generator = self.partition(data, chunk_size=chunk_size)
+
+        total_number_of_expected_results = math.ceil(data_length / chunk_size)
+
+        map_kwargs = {"map_function": map_function, "kwargs": function_kwargs}
+        result = tqdm(self.distribute(_function_with_partly_reduce, chunk_generator, map_kwargs),
+                      total=total_number_of_expected_results,
+                      desc=self.progressbar_title, disable=self.disable_progressbar)
+
+        result = list(itertools.chain.from_iterable(result))
+
+        return result
+
+    def distribute(self, func, partitioned_chunks, kwargs):
+        """
+        This abstract base function distributes the work among workers, which can be threads or nodes in a cluster.
+        Must be implemented in the derived classes.
+        
+        :param func: the function to send to each worker.
+        :type func: callable
+        :param partitioned_chunks: The list of data chunks - each element is again
+            a list of chunks - and should be processed by one worker.
+        :type partitioned_chunks: iterable 
+        
+        :return: The result of the calculation as a list - each item should be the result of the application of func 
+            to a single element.
+        """
+        raise NotImplementedError
+
+    def close(self):
+        """
+        Abstract base function to clean the Distributor after use, e.g. close the connection to a DaskScheduler
+        """
+        pass
+
+
+class MapDistributor(Distributor):
+    """
+    Distributor using the python build-in map, which calculates each job sequentially one after the other.
+    """
+    def distribute(self, func, partitioned_chunks, kwargs):
+        return map(partial(func, **kwargs), partitioned_chunks)
+
+
+class LocalDaskDistributor(Distributor):
+    """
+    Distributor using a local dask cluster and inproc communication.
+    """
+    def __init__(self, n_workers, disable_progressbar, progressbar_title):
+        Distributor.__init__(self, n_workers, disable_progressbar, progressbar_title)
+
+        from distributed import LocalCluster, Client
+        cluster = LocalCluster(n_workers=self.n_workers, processes=True)
+        self.client = Client(cluster)
+
+    def distribute(self, func, partitioned_chunks):
+        result = self.client.gather(self.client.map(func, partitioned_chunks))
+        return result
+
+    def close(self):
+        self.client.close()
+
+
+class ClusterDaskDistributor(Distributor):
+    """
+    Distributor using a dask cluster, meaning that the calculation is spread over a cluster
+    """
+    def __init__(self, n_workers, disable_progressbar, progressbar_title, address):
+        Distributor.__init__(self, n_workers, disable_progressbar, progressbar_title)
```

Comment:
the n_workers has no meaning here, or? So better not use it as a constructor argument and set it to None or something.